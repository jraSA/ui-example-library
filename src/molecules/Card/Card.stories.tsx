import React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import Card from '.';
import MySimpleHelloComponent from '../../atoms/MySimpleHelloComponent';

export default {
    title: 'Molecules/Card',
    component: Card,
} as Meta;

const text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit,"
    + "sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.";

const Template = (args) => (
    <Card {...args}>
        <MySimpleHelloComponent message={text} />
    </Card>
);

export const SimpleCardWithoutTitle = Template.bind({});


export const SimpleCardWithTitle = Template.bind({});
SimpleCardWithTitle.args = { title: 'My title' };