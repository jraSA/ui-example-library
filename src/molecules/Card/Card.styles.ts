import styled from 'styled-components';
import BaseTheme from '../../theme/BaseTheme';
import { getThemeValue } from '../../theme/ThemeUtils';

const { space, padding, sizes, radii, fonts, color } = BaseTheme;

const { lightboxBackground, black, white } = color;
const { primaryFont, fontSizes } = fonts;

const { card } = sizes;

export const CardFrame = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: ${card.width};
  height: ${card.height}px;
  padding: ${padding[4]}px;
  border-radius: ${radii[3]}px;
  background-color: ${(props) =>
    getThemeValue(props, 'primaryBackground', white)};
  box-shadow: ${space[1]}px ${space[1]}px ${space[2]}px ${lightboxBackground};
  &:hover {
    box-shadow: ${space[2]}px ${space[2]}px ${space[3]}px ${lightboxBackground};
  }
`;

export const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Title = styled.h1`
  font-family: ${primaryFont}, arial, sans-serif;
  font-size: ${(props) => getThemeValue(props, 'titleSize', fontSizes[8])}px;
  text-align: left;
  color: ${(props) => getThemeValue(props, 'titleColor', black)};
`;

export const Content = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
