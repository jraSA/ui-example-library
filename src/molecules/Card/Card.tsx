import React from 'react';
import CenteredContainer from '../../atoms/CenteredContainer';
import { CardFrame, Title } from './Card.styles';

interface Props {
    title?: string,
    children?: React.ReactNode
}

const Card = (props: Props) => {
    const { children, title } = props;
    return (
        <CenteredContainer>
            <CardFrame>
                {renderTitle(title)}
                <CenteredContainer>
                    {children}
                </CenteredContainer>
            </CardFrame>
        </CenteredContainer>
    )
};

export default Card;

const renderTitle = (title: string | undefined): React.ReactNode => {
    return title && <Title>{title}</Title>;
};
