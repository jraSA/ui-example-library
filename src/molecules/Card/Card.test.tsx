import React from 'react';
import renderer from 'react-test-renderer';
import Card from '.';

describe('Render test', () => {
    const wrapper = renderer.create(<Card></Card>);

    test('WHEN I render the component THEN it should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

describe('Render test with child', () => {
    const wrapper = renderer.create(<Card><div>Foo-boo</div></Card>);

    test('WHEN I render the component THEN it should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});


describe('Render test with title', () => {
    const wrapper = renderer.create(<Card title={'Foo-title'}><div>Foo-boo</div></Card>);

    test('WHEN I render the component THEN it should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});