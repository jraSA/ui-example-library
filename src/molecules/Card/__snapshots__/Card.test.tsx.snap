// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render test WHEN I render the component THEN it should match the snapshot 1`] = `
<div
  className="sc-bdfBwQ ixcnvM"
>
  <div
    className="sc-gsTCUz fFutaN"
  >
    <div
      className="sc-bdfBwQ ixcnvM"
    />
  </div>
</div>
`;

exports[`Render test with child WHEN I render the component THEN it should match the snapshot 1`] = `
<div
  className="sc-bdfBwQ ixcnvM"
>
  <div
    className="sc-gsTCUz fFutaN"
  >
    <div
      className="sc-bdfBwQ ixcnvM"
    >
      <div>
        Foo-boo
      </div>
    </div>
  </div>
</div>
`;

exports[`Render test with title WHEN I render the component THEN it should match the snapshot 1`] = `
<div
  className="sc-bdfBwQ ixcnvM"
>
  <div
    className="sc-gsTCUz fFutaN"
  >
    <h1
      className="sc-dlfnbm jWLUWq"
    >
      Foo-title
    </h1>
    <div
      className="sc-bdfBwQ ixcnvM"
    >
      <div>
        Foo-boo
      </div>
    </div>
  </div>
</div>
`;
